actions/user_actions.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/local/sbin/charm-env python3
+import json
+import os
+import sys
+from base64 import b64decode
+from charmhelpers.core import hookenv
+from charmhelpers.core.hookenv import (
+    action_get,
+    action_set,
+    action_fail,
+    action_name
+)
+from charms import layer
+
+os.environ['PATH'] += os.pathsep + os.path.join(os.sep, 'snap', 'bin')
+
+# Default namespace for all CK secrets
+NS = 'auth-webhook'
+
+# Import charm layers and start reactive
+layer.import_layer_libs()
+hookenv._run_atstart()
+
+
+def protect_resources(name):
+    '''Do not allow the action to operate on names used by Charmed Kubernetes.'''
+    protected_names = ['admin',
+                       'system:kube-controller-manager', 'kube-controller-manager',
+                       'system:kube-proxy', 'kube-proxy',
+                       'system:kube-scheduler', 'kube-scheduler',
+                       'system:monitoring']
+    if name.startswith('kubelet') or name in protected_names:
+        action_fail('Not allowed to {} "{}".'.format(action, name))
+        sys.exit(0)
+
+
+def user_list():
+    '''Return a dict of 'username: secret_id' for Charmed Kubernetes users.'''
+    output = layer.kubernetes_common.kubectl(
+        '-n', NS, 'get', 'secrets', '-o', 'json').decode('UTF-8')
+    secrets = json.loads(output)
+    users = {}
+    if 'items' in secrets:
+        for secret in secrets['items']:
+            try:
+                secret_id = secret['metadata']['name']
+                username_b64 = secret['data']['username'].encode('UTF-8')
+            except (KeyError, TypeError):
+                # CK secrets will have populated 'data', but not all secrets do
```

Comment:
This is great -- i didn't actually think to check if other types would be more fitting.  This is doubly great because i namespaced all our secrets just to keep them together.  I can drop all the NS code and just `juju.is/secret` or whatever.